File: searchIndex-platform/module/search-api/search-manager/app/controllers/HealthCheckController.java
package controllers;

import org.ekstep.common.dto.Request;

import managers.PlaySearchManager;
import play.libs.F.Promise;
import play.mvc.Result;

public class HealthCheckController extends SearchBaseController {
	
	PlaySearchManager mgr = new PlaySearchManager();

	public Promise<Result> search() {
		String apiId = "search-service.health";
		Request request = getRequest(null, apiId, request().uri());
		return mgr.health(request);
	}
	
}

File: searchIndex-platform/module/search-api/search-manager/app/controllers/HealthCheckController.scala
package controllers

import akka.actor.{ActorRef, ActorSystem}
import javax.inject.{Inject, Named}
import play.api.mvc.ControllerComponents
import managers.PlaySearchManager

import scala.concurrent.ExecutionContext

class HealthCheckController@Inject()(@Named("HealthCheckManager") actor: ActorRef, cc: ControllerComponents, actorSystem: ActorSystem)(implicit exec: ExecutionContext) extends BaseController(cc) {

    val mgr = new PlaySearchManager

    def health() = Action.async { implicit request =>
        val apiId = "search-service.health"
        val internalReq = getRequest(apiId)
        getResult(mgr.health(internalReq, actor))
    }
}
